roles/deploy_olm_operator/tasks/main.yml

```
@@ -1,206 +1,241 @@
+      ignore_errors: true
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+      when: csv_describe_result.stdout != ""
+
+    - name: "Output the operator pod log to a debug file"
+      copy:
+        content: "{{ operator_pod_result.stdout }}"
+        dest: "{{ work_dir }}/olm-operator-pod-debug.txt"
+      when:
+        - csv_describe_result.stdout != ""
+        - operator_pod_result.stdout != ""
+
+    - name: "Get the pod container logs of the deployed operator"
+      shell: "{{ oc_bin_path }} get --output=name pods | grep {{ operator_pod_name }} | cut -d' ' -f1 | xargs -I{} {{ oc_bin_path }} logs {} -c {{ operator_container_name }}"
+      register: operator_container_result
+      ignore_errors: true
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+      when: csv_describe_result.stdout != ""
+
+    - name: "Output the operator container log to a debug file"
+      copy:
+        content: "{{ operator_container_result.stdout }}"
+        dest: "{{ work_dir }}/olm-operator-container-debug.txt"
+      when:
+        - csv_describe_result.stdout != ""
+        - operator_container_result.stdout != ""
+
+    - name: "Get the catalogsource pod log"
+      shell: "{{ oc_bin_path }} get --output=name pods -n openshift-marketplace | grep test-operators | cut -d' ' -f1 | xargs -I{} {{ oc_bin_path }} logs {} -n openshift-marketplace"
+      register: catalog_source_result
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+
+    - name: "Output the catalogsource log to a debug file"
+      copy:
+        content: "{{ catalog_source_result.stdout }}"
+        dest: "{{ work_dir }}/olm-catalog-source-debug.txt"
+      when: catalog_source_result.stdout is defined
+
+    - name: "Get the catalog-operator pod log"
+      shell: "{{ oc_bin_path }} get --output=name pods -n openshift-operator-lifecycle-manager | grep catalog-operator | cut -d' ' -f1 | xargs -I{} {{ oc_bin_path }} logs {} -n openshift-operator-lifecycle-manager | tail -n 100"
+      register: catalog_operator_result
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+
+    - name: "Output the catalog-operator log to a debug file"
+      copy:
+        content: "{{ catalog_operator_result.stdout }}"
+        dest: "{{ work_dir }}/olm-catalog-operator-debug.txt"
+      when: catalog_operator_result.stdout is defined
+
+    - name: "Get the InstallPlan debug output"
+      shell: "{{ oc_bin_path }} describe installplan -n {{ openshift_namespace }}"
+      register: installplan_result
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+
+    - name: "Output the InstallPlan debug output to a debug file"
+      copy:
+        content: "{{ installplan_result.stdout }}"
+        dest: "{{ work_dir }}/olm-installplan-debug.txt"
+      when: installplan_result.stdout is defined
+    
+    - name: "Set fact for deploy olm operator results"
+      set_fact:
+        deploy_olm_operator_results: {}
```

Don't we want to fill out the results json like in the [parsing role](https://github.com/redhat-operator-ecosystem/operator-test-playbooks/blob/master/roles/parse_operator_metadata/tasks/main.yml#L174)? e.g.
```suggestion
        deploy_olm_operator_results: {'result': 'pass', 'msg': ''}
```